Add onElementPositionChange event to SetElementPosition for anti-cheat compatibility

frqher opened the pull request:
This PR adds a server-side "onElementPositionChange" event to the SetElementPosition function.

**Reason:**
Some anti-cheat scripts detect large position changes as cheating, but do not distinguish between server-initiated and client-initiated changes. This causes false positives when the server teleports players (e.g., for elevators or scripted events).

**How it works:**
By triggering the "onElementPositionChange" event server-side, anti-cheat scripts can listen for this event and safely ignore position changes that originate from the server.

This improves compatibility with anti-cheat resources and prevents legitimate server-side teleports from being flagged as cheats.

frqher committed:
Add onElementPositionChange event to SetElementPosition for anti-cheat compatibility

This event allows anti-cheat scripts to distinguish between server-initiated and client-initiated position changes, preventing false positives when the server teleports players (e.g., for elevators or scripted events).

MohabCodeX commented:
I’m not sure, but isn’t [onPlayerTeleport](https://wiki.multitheftauto.com/wiki/OnPlayerTeleport) enough?
But maybe `onElementPositionChange` would be useful for general cases to deal with `SetElementPosition`

frqher commented:
> I’m not sure, but isn’t [onPlayerTeleport](https://wiki.multitheftauto.com/wiki/OnPlayerTeleport) enough? But maybe `onElementPositionChange` would be useful for general cases to deal with `SetElementPosition`

onPlayerTeleport is indeed sufficient for detecting sudden (teleport-like) position changes, especially those initiated by the client.
However, an event like onElementPositionChange can be useful for more general cases, as it allows you to detect all position changes made via server-side scripts (e.g., when using setElementPosition). This is not limited to players, but can also apply to vehicles, peds, objects, etc.

tederis commented:
Gamemode scripts shall not allow cheaters to call server-side setElementPosition. Otherwise it indicates that a gamemode was designed with mistakes. I personally don't like the idea of adding tons of events in trying to fix poorly designed scripts.

blonter commented:
There are multiple issues with `onPlayerTeleport`. It doesn’t trigger when it should, and when it does, it can still produce false positives. More info can be found on Discord. There’s been a lot of discussion about these problems there. 

It’s important to ensure this event only triggers for element position changes initiated on the client side, since we trust the server.

Once everything is fixed, the current `onPlayerTeleport` should be marked as deprecated.

frqher committed:
Add old position parameters to onElementPositionChange event

The onElementPositionChange event now receives both the old and new position of the element as arguments.

frqher commented:
> Gamemode scripts shall not allow cheaters to call server-side setElementPosition. Otherwise it indicates that a gamemode was designed with mistakes. I personally don't like the idea of adding tons of events in trying to fix poorly designed scripts.



<img width="872" height="512" alt="dasd" src="https://github.com/user-attachments/assets/295ed0fd-4e1d-4f54-88c7-15c2af67fd06" />

tzwer commented:
> <img alt="dasd" width="872" height="512" src="https://private-user-images.githubusercontent.com/20355378/482113474-295ed0fd-4e1d-4f54-88c7-15c2af67fd06.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyOTc0NjYsIm5iZiI6MTc1NjI5NzE2NiwicGF0aCI6Ii8yMDM1NTM3OC80ODIxMTM0NzQtMjk1ZWQwZmQtNGUxZC00ZjU0LTg4YzctMTVjMmFmNjdmZDA2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI3VDEyMTkyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwYzE1Y2Y5ZjcwNWI1NjlhNTY5ZjhhNjAwY2UwMzMyYWYyM2RjYTg4NjEwYWM5ZDJkYzI2N2M4NGYwOTdkOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JkAyilhRi-XtskK9aD7UkukCzRBP8HmV3WaMlXZmwi0">

Check these out
- https://github.com/multitheftauto/mtasa-resources/pull/663
- https://github.com/multitheftauto/mtasa-resources/pull/664
